src/pubsub.c

```diff
@@ -615,6 +615,116 @@ void publishCommand(client *c) {
     addReplyLongLong(c,receivers);
 }
 
+typedef struct {
+    dict *subscription_dict;
+    long count;
+    int use_pattern;
+    sds pat;
```

Comment:
You could just use the existence of pat as an indication to use_pattern. Then you can default it properly to NULL.